### PR TITLE
fix: reinstate --all-sub-project support for Gradle

### PR DESCRIPTION
**Pull request by kyegupov**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Fixes `--all-sub-project` flag broken by the Gradle plugin upgrade.
Also, fully use the plugin type definitions from the snyk-cli-interface package.

### CODE REVIEW

File: src/lib/snyk-test/run-test.ts [hunk 1]
```diff
@@ -238,13 +237,13 @@ async function getDepsFromPlugin(root, options: Options): Promise<MultiDepRootsR
...
       plugin: inspectRes.plugin,
-      depRoots: [{depTree: inspectRes.package}],
+      scannedProjects: [{depTree: inspectRes.package}],
     };
   } else {
     // We are using "options" to store some information returned from plugin that we need to use later,
     // but don't want to send to Registry in the Payload.
     // TODO(kyegupov): decouple inspect and payload so that we don't need this hack
-    options.subProjectNames = inspectRes.depRoots.map((depRoot) => depRoot.depTree.name);
+    options.subProjectNames = inspectRes.scannedProjects.map((depRoot) => depRoot.depTree.name);
```
Comment (orsagie) [inline src/lib/snyk-test/run-test.ts]: Change to scannedProject variable name

File: src/lib/snyk-test/run-test.ts [hunk 2]
```diff
@@ -263,7 +262,7 @@ async function assembleLocalPayloads(root, options: Options & TestOptions): Prom
     const deps = await getDepsFromPlugin(root, options);
     analytics.add('pluginName', deps.plugin.name);
 
-    for (const depRoot of deps.depRoots) {
+    for (const depRoot of deps.scannedProjects) {
```
Comment (orsagie) [inline src/lib/snyk-test/run-test.ts]: Same here

File: test/acceptance/cli.acceptance.test.ts [hunk 3]
```diff
@@ -632,7 +632,7 @@ test('`test gradle-app --all-sub-projects` returns correct multi tree meta', asy
     async inspect() {
```
Comment (orsagie) [inline test/acceptance/cli.acceptance.test.ts]: Maybe add the return type here, so we know it's the right one we are building?

File: src/lib/snyk-test/run-test.ts (hunk 1 above)
Comment (kyegupov) [reply src/lib/snyk-test/run-test.ts]: Renamed (+ all mentions of "depRoot" ever)

File: test/acceptance/cli.acceptance.test.ts (hunk 3 above)
Comment (kyegupov) [reply test/acceptance/cli.acceptance.test.ts]: added

**Pull request merged**